### DISCUSSION TITLE
Why are you using Microsoft Search as the Date source?

### DISCUSSION THREAD

**Discussion by kasperbolarsen** [microsoft-search/pnp-modern-search#3258] [General]
I know when and why I am using MS Search as the Data Source in PnP Modern Search:
1) Searching External Items aka MS Graph Connector data
2) Searching Messages
3) Searching Events

I haven't yet found a case where I would prefer to use MS Search over SP Search for Drive, Drive Items, List, List Items and People.

How about you, do you have additional cases where you use MS Search over SP Search?

**Comment by sympmarc** [comment]
FWIW, I don’t think I’ve ever used the Microsoft source. Since all the IA tools are in SharePoint, and all the content using those structures are in SharePoint, using the SharePoint source just makes more sense to me.

**Reply by kasperbolarsen** [reply]
The one exception could be that SP content from the MS search datasource is using a superior ranking model. 🤔

**Reply by sympmarc** [reply]
Yes, but how many people would notice that? And to be honest, I don't know what the superior ranking is. Can you explain or point to any articles?

**Reply by kasperbolarsen** [reply]
If MS are to be believed,  then yes: https://learn.microsoft.com/en-us/graph/search-concept-overview#consistent-up-to-date-search-experience 😁

**Reply by wobba** [reply]
Microsoft do not share internal information on ranking, as it changes, evolves and adapts over time. In general it's graph/signal based as one of many features in the models used 🙂 And personal/contextual ranking outperforms static or equal rank for everyone.

**Reply by sympmarc** [reply]
Since it's a secret, I'm not convinced switching to the Microsoft source is a better way to go. Better the devil I know.

**Reply by wobba** [reply]
All depends on the scenario for search and what you need in terms of ranking. If you need sorting by business logic, then you make up the rules anyways, so one over the other might not be important. Then it ends up on tooling and not features.
